File: lab 6/Binarysearch.py
import math
def Binarysearch(A,e,l,r):
  
   while l<=r:
        m=math.floor((l+r)/2)
        if A[m]==e:
           print("Element found")
           return
        elif A[m]<e :
            l=m+1
        else:
            r=m-1
   print("Element not found")
    


A=[2,3,5,34,65,89]
n=len(A)
e=int(input("Enter the element you want to find"))
Binarysearch(A,e,0,n-1)




    

File: lab 6/binarysearch.c
#include<stdio.h>
#include<math.h>
void Binarysearch(int A[],int l,int r,int e){
  int flag = 0;
   do{
       
       int m = (l + r) / 2;
       if(A[m]==e){
           flag = 1;
           break;
       }
       else if(A[m]>e){
           r = m-1;
       }
       else{
           l = m+1;
          
       }
   } while (l < r);
    if(flag==0){
        printf("element not found");
    }
    else
        printf("Element found");
}
int main(){
    int A[] = {2, 3, 6, 8, 27, 39, 43, 56};
    int n = sizeof(A)/sizeof(A[0]);
    int e;
    printf("Enter the element you want to search");
    scanf("%d", &e);
    Binarysearch(A, 0, n-1 , e);
    return 0;
    }
